omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,331 @@
+		
+		Assert.assertEquals(diagnosis.getUuid(), PropertyUtils.getProperty(result, "uuid"));
+	}
+	
+	@Test
+	public void shouldVoidDiagnosis() throws Exception {
+		
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+		
+		Assert.assertFalse(diagnosis.isVoided());
+		
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", "test");
+		handle(req);
+		
+		diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+		Assert.assertTrue(diagnosis.isVoided());
+		Assert.assertEquals("test", diagnosis.getVoidReason());
+		Assert.assertNotNull(diagnosis.getDateVoided());
+		Assert.assertNotNull(diagnosis.getVoidedBy());
+		Assert.assertEquals(Context.getAuthenticatedUser().getUuid(), diagnosis.getVoidedBy().getUuid());
+	}
+	
+	@Test
+	public void shouldUnVoidDiagnosis() throws Exception {
+		
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(RestTestConstants2_2.VOIDED_DIAGNOSIS_UUID);
+		
+		Assert.assertTrue(diagnosis.isVoided());
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("voided", false);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + RestTestConstants2_2.VOIDED_DIAGNOSIS_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		diagnosis = diagnosisService.getDiagnosisByUuid(RestTestConstants2_2.VOIDED_DIAGNOSIS_UUID);
+		
+		Assert.assertFalse(diagnosis.isVoided());
+		Assert.assertNull(diagnosis.getDateVoided());
+		Assert.assertNull(diagnosis.getVoidedBy());
+		Assert.assertNull(diagnosis.getVoidReason());
+		
+	}
+	
+	@Test
+	public void shouldUpdateANonCodedDiagnosis() throws Exception {
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(RestTestConstants2_2.UPDATABLE_NON_CODED_DIAGNOSIS_UUID);
+		
+		Assert.assertFalse(diagnosis.getVoided());
+		Assert.assertNull(diagnosis.getDiagnosis().getCoded());
+		Assert.assertEquals((int) diagnosis.getRank(), 2);
+		Assert.assertEquals(diagnosis.getCondition().getUuid(), "e804ee60-ecbc-4d70-abda-1e4f6f64e5b5");
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
+		
+		codedOrFreeText.add("coded", concept.getUuid());
+		codedOrFreeText.add("specificName", conceptName.getUuid());
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("voided", true);
+		attributes.add("diagnosis", codedOrFreeText);
+		attributes.add("condition", condition.getUuid());
+		attributes.add("rank", 1);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/"
+		        + RestTestConstants2_2.UPDATABLE_NON_CODED_DIAGNOSIS_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		Diagnosis new_diagnosis = diagnosisService
+		        .getDiagnosisByUuid(RestTestConstants2_2.UPDATABLE_NON_CODED_DIAGNOSIS_UUID);
+		
+		Assert.assertTrue(new_diagnosis.getVoided());
```

Comment:
Let not this appear as a test for voiding.
It should be about the updatable properties.

Reply:
I included the test for un voiding because it is among the updatable properties.